cmd/ci-secret-bootstrap/main.go

```diff
@@ -175,29 +176,53 @@ func (o *options) validateCompletedOptions() error {
 			if key == "" {
 				return fmt.Errorf("config[%d].from: empty key is not allowed", i)
 			}
-			if bwContext.BWItem == "" {
+
+			if bwContext.BWItem == "" && len(bwContext.DockerConfigJSONData) == 0 {
 				return fmt.Errorf("config[%d].from[%s]: empty value is not allowed", i, key)
 			}
-			switch bwContext.Attribute {
-			case secretbootstrap.AttributeTypePassword, "":
-			default:
-				return fmt.Errorf("config[%d].from[%s].attribute: only the '%s' is supported, not %s", i, key, secretbootstrap.AttributeTypePassword, bwContext.Attribute)
-			}
-			nonEmptyFields := 0
-			if bwContext.Field != "" {
-				nonEmptyFields++
-			}
-			if bwContext.Attachment != "" {
-				nonEmptyFields++
-			}
-			if bwContext.Attribute != "" {
-				nonEmptyFields++
-			}
-			if nonEmptyFields == 0 {
-				return fmt.Errorf("config[%d].from[%s]: one of [field, attachment, attribute] must be set", i, key)
+
+			if bwContext.BWItem != "" && len(bwContext.DockerConfigJSONData) > 0 {
+				return fmt.Errorf("config[%d].from[%s]: both bitwarden dockerconfigJSON items are not allowed.", i, key)
 			}
-			if nonEmptyFields > 1 {
-				return fmt.Errorf("config[%d].from[%s]: cannot use more than one in [field, attachment, attribute]", i, key)
+
+			if len(bwContext.DockerConfigJSONData) > 0 {
+				for _, data := range bwContext.DockerConfigJSONData {
+					if data.BWItem == "" {
+						return fmt.Errorf("config[%d].from[%s]: bitwarden item is missing", i, key)
+					}
+					if data.RegistryURL == "" {
+						return fmt.Errorf("config[%d].from[%s]: registry_url field is missing", i, key)
+					}
+					if data.Auth == "" {
+						return fmt.Errorf("config[%d].from[%s]: auth field is missing", i, key)
+					}
+
+					if secretConfig.To[i].Type != "kubernetes.io/dockerconfigjson" {
```

Comment:
Shouldn't we default it first if it is empty?

Reply:
Sure.

Reply:
addressed